strawberry/field.py

```diff
@@ -105,11 +122,51 @@ def __call__(self, resolver: _RESOLVER_TYPE) -> "StrawberryField":
 
     @property
     def arguments(self) -> List[StrawberryArgument]:
+        arguments_as_annotations = self.get_arguments()
+
         if not self.base_resolver:
             return []
 
+        # Convert the arguments to StrawberryArgument types
+        arguments = []
+        for arg_name, annotation in arguments_as_annotations.items():
+            if isinstance(annotation, BareStrawberryArgument):
+                argument = annotation
+            else:
+                default = self.base_resolver.get_argument_default(arg_name)
+                argument = self.create_argument(annotation, default)
+
+            arguments.append(argument.as_argument(arg_name))
```

Comment:
It might be a bit confusing to call the mapping values here `annotation`s. `StrawberryField.get_arguments` returns `Dict[str, object]` not `Dict[str, StrawberryAnnotation]`.